streams/src/main/java/org/apache/kafka/streams/processor/internals/ThreadMetadataImpl.java

```diff
@@ -57,7 +57,11 @@ public ThreadMetadataImpl(final String threadName,
                               final Set<TaskMetadata> standbyTasks) {
         this.mainConsumerClientId = mainConsumerClientId;
         this.restoreConsumerClientId = restoreConsumerClientId;
-        this.producerClientIds = Collections.unmodifiableSet(producerClientIds);
+        if (producerClientIds != null) {
+            this.producerClientIds = Collections.unmodifiableSet(producerClientIds);
+        } else {
+            this.producerClientIds = Collections.emptySet();
+        }
```

Comment:
@cadonna, I needed to add this guard as ThreadMetadataTest was failing with NPE.